x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/comparison_side/subfield_header.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { camelCase, startCase } from 'lodash';
+import { EuiTitle, EuiSpacer } from '@elastic/eui';
+import { fieldToDisplayNameMap } from '../../diff_components/translations';
+import type { DiffableAllFields } from '../../../../../../../common/api/detection_engine';
+
+function shouldDisplaySubfieldLabelsForField(fieldName: string): boolean {
+  return [
+    'data_source',
+    'kql_query',
+    'eql_query',
+    'esql_query',
+    'threat_query',
+    'rule_schedule',
+    'rule_name_override',
+    'timestamp_override',
+    'timeline_template',
+    'building_block',
+    'threshold',
+  ].includes(fieldName);
+}
+
+interface SubfieldHeaderProps {
+  fieldName: keyof DiffableAllFields;
+  subfieldName: string;
+}
+
+export function SubfieldHeader({ fieldName, subfieldName }: SubfieldHeaderProps) {
+  if (!shouldDisplaySubfieldLabelsForField(fieldName)) {
```

Comment:
`SubfieldHeader` as a leaf component shouldn't know about implementation details of the other components. This logic should be a few levels up in `Subfield` component. It's also good to leave a comment why header isn't always shown.

Reply:
Done.